uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-calendar_test.js

```diff
@@ -1,30 +1,412 @@
+            });
+
+            expect(returnValue).toEqual(filteredContacts);
+          });
+        });
+
+      });
+
+      describe('refresh', function () {
+        it('page loading is true initially', function () {
+          controller.refresh();
+          expect(controller.loading.calendar).toBe(true);
+        });
+
+        describe('after data load is complete', function () {
+          beforeEach(function () {
+            controller.refresh();
+            $rootScope.$digest();
+          });
+
+          it('page loading is false', function () {
+            expect(controller.loading.calendar).toBe(false);
+          });
+        });
+
+        describe('isWeekend', function () {
+          var dateObj;
+
+          beforeEach(function () {
+            controller.refresh();
+            $rootScope.$digest();
+          });
+
+          it('is set', function () {
+            _.each(controller.managedContacts, function (contact) {
+              dateObj = contact.calendarData.days[getDateObjectWithFormat(getDate('weekend').date).valueOf()];
+              expect(dateObj.UI.isWeekend).toBe(true);
+            });
+          });
+        });
+
+        describe('isNonWorkingDay', function () {
+          var dateObj;
+
+          beforeEach(function () {
+            controller.refresh();
+            $rootScope.$digest();
+          });
+
+          it('is set', function () {
+            _.each(controller.managedContacts, function (contact) {
+              dateObj = contact.calendarData.days[getDateObjectWithFormat(getDate('non_working_day').date).valueOf()];
+              expect(dateObj.UI.isNonWorkingDay).toBe(true);
+            });
+          });
+        });
+
+        describe('isPublicHoliday', function () {
+          var dateObj;
+
+          beforeEach(function () {
+            spyOn(controller, 'isPublicHoliday').and.returnValue(true);
+            controller.refresh();
+            $rootScope.$digest();
+          });
+
+          it('is set', function () {
+            _.each(controller.managedContacts, function (contact) {
+              dateObj = contact.calendarData.days[Object.keys(contact.calendarData.days)[0]];
+              expect(dateObj.UI.isPublicHoliday).toBe(true);
+            });
+          });
+        });
+
+        describe('refresh', function () {
+          var dateObj,
+            workPattern,
+            leaveRequest;
+
+          beforeEach(function () {
+            workPattern = workPatternData.daysData();
+            leaveRequest = leaveRequestData.singleDataSuccess().values[0];
+            workPattern.values[0].date = leaveRequest.from_date;
+          });
+
+          describe('when leave request is not approved', function () {
+            beforeEach(function () {
+              var status = _.find(optionGroupMock.getCollection('hrleaveandabsences_leave_request_status'), function (status) {
+                return status.name === 'waiting_approval';
+              });
```

Comment:
is there an helper method to get/filter option value by name on `optionGroupMock`?